### ISSUE TITLE
BG2: Cannot start new game or load some games (assertion error)

### ISSUE THREAD

**Issue by MarcelHB** [gemrb/gemrb#1675]
#### Bug description
I've only tested original BG 2 + ToB while trying to look at #1674. Clicking some random PC for a new game, the game reproducibly fails to enter a game after loading. I also have a save game (Irenicus' dungeon) that appears to hit the same error (although not sure how old, another one looks good).

Problem appears to be related To Rielev?

#### Steps to reproduce
1. Try to start a new game in BG2.

#### Screenshots
```
Assertion failed: hittingHeader, file D:/Sources/gemrb/gemrb/core/Inventory.cpp, line 1363
Thread 1 hit Breakpoint 1, 0x00007ffd1103f1e7 in msvcrt!abort () from C:\Windows\System32\msvcrt.dll
(gdb) bt
#0  0x00007ffd1103f1e7 in msvcrt!abort () from C:\Windows\System32\msvcrt.dll
#1  0x00007ffd1103da1e in msvcrt!_wassert ()
   from C:\Windows\System32\msvcrt.dll
#2  0x00007ffcb90597e9 in GemRB::Inventory::CacheWeaponInfo (
    this=0x194b854c8e0, leftOrRight=false)
    at D:/Sources/gemrb/gemrb/core/Inventory.cpp:1363
#3  0x00007ffcb90595e9 in GemRB::Inventory::CacheAllWeaponInfo (
    this=0x194b854c8e0) at D:/Sources/gemrb/gemrb/core/Inventory.cpp:1326
#4  0x00007ffcb90594da in GemRB::Inventory::SetEquippedSlot (
    this=0x194b854c8e0, slotcode=0, header=0, noFX=false)
    at D:/Sources/gemrb/gemrb/core/Inventory.cpp:1291
#5  0x00007ffcb90f73f7 in GemRB::Actor::SetEquippedQuickSlot (
    this=0x194b854bc00, slot=0, header=0)
    at D:/Sources/gemrb/gemrb/core/Scriptable/Actor.cpp:8622
#6  0x00007ffcb90e7ffd in GemRB::Actor::SetMap (this=0x194b854bc00,
    map=0x194b82fc410)
    at D:/Sources/gemrb/gemrb/core/Scriptable/Actor.cpp:4681
#7  0x00007ffcb9073070 in GemRB::Map::InitActors (this=0x194b82fc410)
    at D:/Sources/gemrb/gemrb/core/Map.cpp:2054
#8  0x00007ffcb8f90151 in GemRB::Game::LoadMap (this=0x194b2eb3310,
    resRef=..., loadscreen=true) at D:/Sources/gemrb/gemrb/core/Game.cpp:863
#9  0x00007ffcb8f8f5a8 in GemRB::Game::GetMap (this=0x194b2eb3310,
    areaname=..., change=true) at D:/Sources/gemrb/gemrb/core/Game.cpp:685
#10 0x00007ffcb8f34d66 in GemRB::GameControl::ChangeMap (this=0x194b57f2d70,
    pc=0x194b524a250, forced=true)
    at D:/Sources/gemrb/gemrb/core/GUI/GameControl.cpp:2494
#11 0x00007ffcb901f673 in GemRB::Interface::HandleFlags (this=0x194a1c57560)
    at D:/Sources/gemrb/gemrb/core/Interface.cpp:482
#12 0x00007ffcb90202aa in GemRB::Interface::Main (this=0x194a1c57560)
    at D:/Sources/gemrb/gemrb/core/Interface.cpp:632
#13 0x00007ff71a8216c2 in main (argc=3, argv=0x194a1c72540)
    at D:/Sources/gemrb/platforms/windows/GemRB.cpp:53
(gdb) f 2
#2  0x00007ffcb90597e9 in GemRB::Inventory::CacheWeaponInfo (
    this=0x194b854c8e0, leftOrRight=false)
    at D:/Sources/gemrb/gemrb/core/Inventory.cpp:1363
1363            assert(hittingHeader);
(gdb) p *item
$1 = {ext_headers = std::vector of length 0, capacity 0,
  equipping_features = std::vector of length 1, capacity 1 = {0x194b8554ec0},
  Name = {str = "MAGE01\000\000"}, ItemName = 6348,
  ItemNameIdentified = 6835, ReplacementItem = {
    str = "\000\000\000\000\000\000\000\000"}, Flags = 104, ItemType = 10,
  UsabilityBitmask = 0, AnimationType = {str = "\000\000"},
  MinLevel = 0 '\000', unknown1 = 0 '\000', MinStrength = 0 '\000',
  unknown2 = 0 '\000', MinStrengthBonus = 0 '\000',
  MinIntelligence = 0 '\000', MinDexterity = 0 '\000', MinWisdom = 0 '\000',
  MinConstitution = 0 '\000', WeaProf = 255 '▒', MinCharisma = 0 '\000',
  unknown3 = 0 '\000', KitUsability = 0, Price = 1, MaxStackAmount = 0,
  ItemIcon = {str = "IRING05\000"}, LoreToID = 0, GroundIcon = {
    str = "GRING01\000"}, Weight = 0, ItemDesc = 2969,
  ItemDescIdentified = GemRB::ieStrRef::INVALID, DescriptionIcon = {
    str = "CRING05\000"}, Enchantment = 0, ExtHeaderOffset = 114,
  FeatureBlockOffset = 114, EquippingFeatureOffset = 0,
  EquippingFeatureCount = 1, Dialog = {
    str = "*\000\000\000\000\000\000\000"}, DialogName = 0,
  WieldColor = 65535, unknown = '\000' <repeats 25 times>, ItemExcl = 0}
(gdb) f 6
#6  0x00007ffcb90e7ffd in GemRB::Actor::SetMap (this=0x194b854bc00,
    map=0x194b82fc410)
    at D:/Sources/gemrb/gemrb/core/Scriptable/Actor.cpp:4681
4681                    SetEquippedQuickSlot(inventory.GetEquipped(), inventory.GetEquippedHeader());
(gdb) p *this
$2 = {<GemRB::Movable> = {<GemRB::Selectable> = {<GemRB::Scriptable> = {
        _vptr.Scriptable = 0x7ffcb91d0710 <vtable for GemRB::Actor+16>,
        WaitCounter = 0, script_timers = std::map with 0 elements,
        globalID = 10127, triggers = std::__cxx11::list = {[0] = {
            triggerID = 54, param1 = 0, param2 = 0, flags = 1}},
        area = 0x194b82fc410, scriptName = {
          str = "rielev", '\000' <repeats 26 times>, static Size = 32 ' '},
        InternalFlags = 4524544, Dialog = {str = "rielev\000\000",
          static npos = 255 '▒'}, actionQueue = empty std::__cxx11::list,
        CurrentAction = 0x0, overHeadTextPos = {x = -1, y = -1},
        overheadTextDisplaying = false, timeStartDisplaying = 0,
...
```

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.0
- [ ] 0.8.8

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [x] SDL2 built with `USE_OPENGL_BACKEND`
- [ ] SDL2 without `USE_OPENGL_BACKEND`

**Comment by lynxlynxlynx** [comment]
I have one of these saves as well. Rielev, the jar creature, has a ring of invisibility for some unknown reason and in an unfortunate slot. I couldn't find any reference anywhere of he getting it. I thought I transplanted some mod save or sth, but if two of us have it, perhaps there's more to it.

Now I got the idea to check the fixpack and indeed this is fixed by it:
https://github.com/Gibberlings3/BG2-Fixpack/blob/4e47a5cdc147c2541515c64ca97684fd4806aca3/bg2fixpack/lib/core_creature_inventory.tph#L1197-L1198

So I'm inclined to just close this, since we do require the fixpack. He has 0 for Equipped, but we could also do more checks if items are sensible (EquipItem, SetEquippedQuickSlot?) somehow. The games do allow for item type mismatches, so I'm not sure if it would break some other assumption.
